lms/djangoapps/student_account/views.py

```diff
@@ -119,6 +119,11 @@ def login_and_registration_form(request, initial_mode="login"):
         'disable_footer': True,
     }
 
+    origin = request.GET.get('origin')
+    if origin and (settings.CORS_ORIGIN_ALLOW_ALL or origin in \
```

Comment:
- Break this into separate statements.
- Tag your changes with `# Edraak:` comment to make it easy to cherry-pick later and to document this change now..